From 25f5e98de88d234e8b33e312c92155bee24c17a3 Mon Sep 17 00:00:00 2001
From: Andre Klapper <a9016009@gmx.de>
Date: Mon, 26 Aug 2024 23:32:52 +0200
Subject: [PATCH 06/14] Fix PHP 8.1 "strlen(null)" exception in
 HTTPFutureHTTPResponseStatus

Summary:
`strlen()` was used in Phabricator to check if a generic value is a non-empty string.
This behavior is deprecated since PHP 8.1. Phorge adopts `phutil_nonempty_string()` as a replacement.

Note: this may highlight other absurd input values that might be worth correcting
instead of just ignoring. If phutil_nonempty_string() throws an exception in your
instance, report it to Phorge to evaluate and fix that specific corner case.

```
ERROR 8192: strlen(): Passing null to parameter #1 ($string) of type string is deprecated at [/var/www/html/phorge/arcanist/src/future/http/status/HTTPFutureHTTPResponseStatus.php:16]
```

Closes T15930

Test Plan: Run `arcanist/bin/arc unit --everything` on a PHP >= 8.1 system.

Reviewers: O1 Blessed Committers, valerio.bozzolan

Reviewed By: O1 Blessed Committers, valerio.bozzolan

Subscribers: tobiaswiese, valerio.bozzolan, Matthew, Cigaryno

Maniphest Tasks: T15930

Differential Revision: https://we.phorge.it/D25810
---
 src/future/http/status/HTTPFutureHTTPResponseStatus.php | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/src/future/http/status/HTTPFutureHTTPResponseStatus.php b/src/future/http/status/HTTPFutureHTTPResponseStatus.php
index 44636dca8ad8..a63130fd056f 100644
--- a/src/future/http/status/HTTPFutureHTTPResponseStatus.php
+++ b/src/future/http/status/HTTPFutureHTTPResponseStatus.php
@@ -11,6 +11,8 @@ final class HTTPFutureHTTPResponseStatus extends HTTPFutureResponseStatus {
     array $headers,
     $expect = null) {
 
+    // As $body can be null (T15930), make sure this code deals with a string
+    $body = phutil_string_cast($body);
     // NOTE: Avoiding PhutilUTF8StringTruncator here because this isn't lazy
     // and responses may be large.
     if (strlen($body) > 512) {
-- 
2.48.1

